extensions/applicationinsights-cfgsync-js/src/CfgSyncHelperFuncs.ts

```diff
@@ -39,3 +42,82 @@ export function replaceByNonOverrideCfg<T=IConfiguration & IConfig, T1=NonOverri
     // if errors happen, do nothing
     return cfg;
 }
+
+//                                cdn Mode (cdn Value = B, SDK Config Defaults = C)
+//                   |--------------------------------------------------------------------------|
+//                   |                    | none        | disabled    | enabled     | force     |
+//                   | ------------------ | ----------- | ----------- | ----------- | --------- |
+// | User Mode       | none               | A || C      | B || A || C | B || A || C | B || C    |
+// (user Value = A)  | disabled           | A || C      | B || A || C | A || C      | B || C    |
+//                   | enabled            | A || C      | B || A || C | A || B || C | B || C    |
+//                   | none(blockCdn)     | A || C      | A || C      | A || C      | A || C    |
+//                   | disabled(blockCdn) | A || C      | A || C      | A || C      | A || C    |
+//                   | enabled(blockCdn)  | A || C      | A || C      | A || C      | A || C    |
+export function shouldOptInFeature(field: string, cdnCfg?: ICfgSyncCdnConfig, customOptInDetails?: IFeatureOptIn) {
+    
+    if (!cdnCfg || !cdnCfg.enabled || !cdnCfg.featureOptIn || !cdnCfg.featureOptIn[field] || !customOptInDetails || !customOptInDetails[field]) {
```

Comment:
I think this isn't quite correct...

for example, if the user (or SDK defaults) doesn't provide any value (undefined) for `customOptInDetails[field]` then this wouldn't allow any CDN value to be validated (I think)